cmd/progress-bar.go

```diff
@@ -102,18 +96,17 @@ func (p *progressBar) Set64(length int64) *progressBar {
 	return p
 }
 
-func (p *progressBar) Seek(offset int64, whence int) (n int64, err error) {
-	switch whence {
-	case io.SeekStart:
-		p.ProgressBar = p.ProgressBar.Set64(offset)
-		return offset, nil
-	case io.SeekCurrent:
-		return p.ProgressBar.Add64(offset), nil
-	case io.SeekEnd:
-		p.ProgressBar.Set64(p.ProgressBar.Total - offset)
-		return p.ProgressBar.Total - offset, nil
-	}
-	return 0, errors.New("invalid argument")
+func (p *progressBar) Read(buf []byte) (n int, err error) {
+	defer func() {
+		// After updating the internal progress bar, make sure that its
+		// current progress doesn't exceed the specified total progress
+		currentProgress := p.ProgressBar.Get()
+		if currentProgress > p.ProgressBar.Total {
+			p.ProgressBar.Set64(p.ProgressBar.Total)
```

Comment:
I think this logic has a problem: After so many retries, we'll reach the total but we are not going to be completely done yet. That means, end-user will see the same total number for sometime while the copy is still continuing and the displayed number will show the total size until the copy is  really done...

Reply:
> I think this logic has a problem: After so many retries, we'll reach the total but we are not going to be completely done yet. That means, the progress bar will be still moving forward, while the displayed number is stuck at the total size for a while until the copy is really done.

mc will still show an error when the upload fails even after having the progress bar arrived to 100%. 

This actually can happen, for example, during complete multipart upload.

In the future, we can make the progress bar reduces its progress when a part upload fails